sonar-dotnet-shared-library/src/test/java/org/sonarsource/dotnet/shared/plugins/AbstractLanguageConfigurationTest.java

```diff
@@ -73,6 +73,36 @@ public void vulnerabilityCategories_reads_configuration() {
     assertThat(config.vulnerabilityCategories()).containsExactly("A", "B", "C");
   }
 
+  @Test
+  public void analyzeGeneratedCodeIsTrue() {
+    Configuration configuration = mock(Configuration.class);
+    when(configuration.getBoolean("sonar.cs.analyzeGeneratedCode")).thenReturn(Optional.of(true));
+
+    AbstractLanguageConfiguration config = createConfiguration(configuration);
+
+    assertThat(config.analyzeGeneratedCode()).isTrue();
+  }
+
+  @Test
+  public void analyzeGeneratedCodeIsFalse() {
+    Configuration configuration = mock(Configuration.class);
+    when(configuration.getBoolean("sonar.cs.analyzeGeneratedCode")).thenReturn(Optional.of(false));
+
+    AbstractLanguageConfiguration config = createConfiguration(configuration);
+
+    assertThat(config.analyzeGeneratedCode()).isFalse();
+  }
+
+  @Test
+  public void analyzeGeneratedCodeIsEmpty() {
```

Comment:
It's not clear what's expected result from the test name. Can we rename it to `analyzeGeneratedCodeIsFalseByDefault`?